node/src/neighborhood/node_location.rs

```diff
@@ -98,37 +91,25 @@ mod tests {
     }
 
     #[test]
-    fn create_gossip_node_with_addr_and_country_code_reveal_results_in_node_with_addr_adn_free_world_bit(
+    fn node_record_from_gossip_with_addr_and_country_is_populated_with_right_addr_and_free_world_bit(
```

Comment:
More importantly, I again think that we're dealing with a redundant test here... 
This test proves exactly the same thing as the one mentioned in the previous comment. It makes no sense to me to test `NodeRecord::try_from(<gossip>)` in this file. It's use of much different code than the simple code this file holds. 

Therefore, believe me or not, I'd like to delete even this one.

Reply:
So you mean we do no t want to test `try_from` for NodeRecord? Moreover this test is actually testing the persistance of `country_code` in Gossip. Do you really wan to delete it?

Reply:
I want you to learn critical thinking. You should find the answers hopefully with just clues given by another person. I didn't do a perfect job writing the comment but I think it still could at least direct you to go checking what I brought up. That is, the act in this test is actually a function call of a function that has nothing to do with this file. I wouldn't think it was untested as it is code that had been around long before you joined the team. Therefore, the easiest thing to do is to go to the file where this function is declared and search for tests that back it. You'll eventually find some. Now, if anything changed about what happens inside the call, it should only be reflected in the assertions of those tests unlike a new one. If the old tests have accommodated any additional things or changed behavior, then you're fine. This is your all concern. 

So before anything else, instead of jumping on writing a new test right away, check the old ones and see if you could enhance them to test all you need. It usually works pretty well.